File: tests/machine/x/rb/cross_join_filter.rb
require 'ostruct'

nums = [1, 2, 3]
letters = ["A", "B"]
pairs = (begin
	_res = []
	for n in nums
		for l in letters
			if ((n % 2) == 0)
				_res << OpenStruct.new(n: n, l: l)
			end
		end
	end
	_res
end)
puts("--- Even pairs ---")
pairs.each do |p|
	puts([p.n, p.l].join(" "))
end

File: tests/machine/x/dart/cross_join_filter.dart
var nums = [1, 2, 3];

var letters = ['A', 'B'];

var pairs = (() {
  var _q0 = <dynamic>[];
  for (var n in nums) {
    for (var l in letters) {
      if (!((n % 2 as int) == 0)) continue;
      _q0.add({'n': n, 'l': l});
    }
  }
  return _q0;
})();

void main() {
  print('--- Even pairs ---');
  for (var p in pairs) {
    print([p['n'], p['l']].join(' '));
  }
}
